perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -989,6 +989,42 @@ public void removeMember(PerunSession sess, Group group, Member member) throws I
 		}
 	}
 
+	@Override
+	public void removeMembers(PerunSession sess, Group group, List<Member> members) throws InternalErrorException, NotGroupMemberException, GroupNotExistsException {
+		// Check if the group is NOT members or administrators group
+		if (group.getName().equals(VosManager.MEMBERS_GROUP)) {
+			throw new InternalErrorException("Cannot remove member directly from the members group.");
+		} else {
+			Collections.sort(members);
+			for (Member member : members) {
+				try {
+					this.removeDirectMember(sess, group, member);
+				} catch (WrongAttributeValueException | WrongReferenceAttributeValueException ex){
+					throw new InternalErrorException(ex);
+				}
+			}
+		}
+	}
+
+
+	@Override
+	public void removeMember(PerunSession sess, List<Group> groups, Member member) throws InternalErrorException, NotGroupMemberException, GroupNotExistsException {
+		Collections.sort(groups);
+		Collections.reverse(groups);
```

Comment:
I think this code is more effective. 
`Collections.sort(groups, Collections.reverseOrder());`